### ISSUE TITLE
How to plot X-Y data series that does not have constant X sample step?

### ISSUE THREAD

**Issue by esutton** [tomauty/react-native-chart#62]
For example 3 data points that are not evenly spaced on the distance X axis; at 0m, 16m, and 139m.

<img width="365" alt="screen shot 2016-03-25 at 12 32 56 pm" src="https://cloud.githubusercontent.com/assets/6115047/14049830/c37359a6-f285-11e5-9132-fb742c13240c.png">

Plot as evenly spaced because the X value is not part of the data series.  This is not what is needed.

<img width="598" alt="screen shot 2016-03-25 at 12 34 20 pm" src="https://cloud.githubusercontent.com/assets/6115047/14049869/037d306c-f286-11e5-8e69-8bf1de8fb9ad.png">

Does your API have a way to plot X and Y values?

| X | Y |
| --- | --- |
| 00.00 | 1.00 |
| 16.13 | 3.00 |
| 139.59 | 6.00 |

It seems like I need to create a data series with 150 null points to cover 150 meters, then round up to nearest, and set the non-null data points at locations 

```
data[0]= 1
data[16] = 3
data[140] = 6.00
```

Thanks in advance,

**Comment by tomauty** [comment]
Right...good question. I'll be looking into this tomorrow. I would consider this a bug.

**Comment by tomauty** [comment]
So I think your solution is what will work on the current master release. I think 2.0's `horizontalGridStep` might take care of this?

**Comment by esutton** [comment]
I like the [ x,y ] pairs being part of the dataset

https://github.com/thunder9/xy.js/tree/master

var datasets = [
  {
    lineColor : 'rgba(220,220,220,1)',
    pointColor : 'rgba(220,220,220,1)',
    pointStrokeColor : '#fff',
    data : [[-4, -2], [-2.5, 1.3], [0, 0], [1, 1.5], [3, 1]]
  },
  {
    lineColor : 'rgba(151,187,205,1)',
    pointColor : 'rgba(151,187,205,1)',
    pointStrokeColor : '#fff',
    data : [[-3, 3], [-1, -1], [0.5, 1], [1.5, -3], [2.8, -1.6]]
  }
];

**Comment by tomauty** [comment]
I'm leaning towards this too. Seems more natural for providing data that you'd...want to chart...

**Comment by esutton** [comment]
Supporting an [ x, y ] data series absolutely makes sense for any practical chart control.

The X ( or time value ) associated with a Y value must be independent of whatever X or Y grid interval the graph uses.

**Comment by tomauty** [comment]
Agreed. That will be the format in 2.0. Also expect that all chart props will be at the base level. No reason for nested props i.e. `chartData` when there's only one data set per visual.

**Comment by tomauty** [comment]
Hi, so I am going to consider this an enhancement for now. Still thinking through how it might work. This is how the horizontal step is currently calculated:

``` javascript
const horizontalStep = containerWidth / data.length;
```

and each data point has its X-value set by

``` javascript
const x = (horizontalStep * i) + horizontalStep
```

The desired behavior, I THINK, would require calculating each X value in relation to a calculated scale, which is how I do Y positions currently.

**Comment by tomauty** [comment]
I think the simplest scale would just be `containerWidth / calculateHighestXValueInSet()`. Does that seem like sound logic? Then I could just do something like

``` javascript
const scale = containerWidth / calculateHighestXValueInSet();
const x = xValue * scale;
```

**Comment by tomauty** [comment]
As a generic FYI — I am no longer able to maintain this library. I recommend checking out [victory-native](https://github.com/FormidableLabs/victory-native) as it's much more maintained.
